File: JS/ParOuImpar.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
 const TEST = require('tape')
 const OBJ = require('./ParOuImpar.js')
 
 TEST('Starting ParOuImpar test...', (t) => {
     //Test 1
     t.assert(OBJ.typeNumberMsg(-5) === "ODD NEGATIVE", "Expect: -5 = ODD NEGATIVE")
     //Test 2
     t.assert(OBJ.typeNumberMsg(0) === "NULL", "Expect: 0 = NULL")
     //Test 3
     t.assert(OBJ.typeNumberMsg(3) === "ODD POSITIVE", "Expect: 3 = ODD POSITIVE")
     //Test 4
     t.assert(OBJ.typeNumberMsg(-4) === "EVEN NEGATIVE", "Expect: -4 = EVEN NEGATIVE")
 
     t.end()
 })
 

File: GO/ParOuImpar_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestParOuImpar(t *testing.T) {
	tables := []struct {
		x    int
		want string
	}{
		{-5, "ODD NEGATIVE"},
		{0, "NULL"},
		{3, "ODD POSITIVE"},
		{-4, "EVEN NEGATIVE"},
	}

	for _, table := range tables {
		got := typeNumberMsg(table.x)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
